WordPress/Classes/ViewRelated/Feature Highlight/TooltipPresenter.swift

```diff
@@ -0,0 +1,159 @@
+            name: UIContentSizeCategory.didChangeNotification, object: nil
+        )
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(recalculatePosition),
+            name: UIDevice.orientationDidChangeNotification,
+            object: nil
+        )
+    }
+
+    func show() {
+        containerView.addSubview(tooltip)
+        self.tooltip.alpha = 0
+        tooltip.addArrowHead(toXPosition: arrowOffsetX(), arrowPosition: tooltipOrientation())
+        setUpConstraints()
+        containerView.layoutIfNeeded()
+
+        UIView.animate(
+            withDuration: Constants.tooltipAnimationDuration,
+            delay: 0,
+            options: .curveEaseOut
+        ) {
+            guard let tooltipTopConstraint = self.tooltipTopConstraint else { return }
+
+            self.tooltip.alpha = 1
+            tooltipTopConstraint.constant -= Constants.tooltipTopConstraintAnimationOffset
+            self.containerView.layoutIfNeeded()
+        }
+    }
+
+    private func configureDismissal() {
+        tooltip.primaryButtonAction = {
+            UIView.animate(
+                withDuration: Constants.tooltipAnimationDuration,
+                delay: 0,
+                options: .curveEaseOut
+            ) {
+                guard let tooltipTopConstraint = self.tooltipTopConstraint else { return }
+
+                self.tooltip.alpha = 0
+                tooltipTopConstraint.constant += Constants.tooltipTopConstraintAnimationOffset
+                self.containerView.layoutIfNeeded()
+            } completion: { isSuccess in
+                self.primaryTooltipAction?()
+                self.tooltip.removeFromSuperview()
+            }
+        }
+    }
+
+    private func setUpConstraints() {
+        tooltip.translatesAutoresizingMaskIntoConstraints = false
+
+        var tooltipConstraints: [NSLayoutConstraint] = [
+            tooltip.centerXAnchor.constraint(equalTo: containerView.centerXAnchor, constant: extraArrowOffsetX())
+        ]
+
+        switch tooltipOrientation() {
+        case .bottom:
+            tooltipTopConstraint = targetView.topAnchor.constraint(
+                equalTo: tooltip.bottomAnchor,
+                constant: Constants.verticalTooltipDistanceToFocus + Constants.tooltipTopConstraintAnimationOffset
+            )
+        case .top:
+            tooltipTopConstraint = tooltip.topAnchor.constraint(
+                equalTo: targetView.bottomAnchor,
+                constant: Constants.verticalTooltipDistanceToFocus + Constants.tooltipTopConstraintAnimationOffset
+            )
+        }
+
+        tooltipConstraints.append(tooltipTopConstraint!)
+        NSLayoutConstraint.activate(tooltipConstraints)
+    }
+
+    @objc private func recalculatePosition() {
+        tooltip.removeFromSuperview()
+        show()
+    }
+
+    /// Calculates where the arrow needs to place in the borders of the tooltia.
```

Comment:
Nice! 😄 Thanks for adding the docs. I've spotted some super minor typos here:

```suggestion
    /// Calculates where the arrow needs to place in the borders of the tooltip.
```